### ISSUE TITLE
Matrix M in objectivity reference frame invariance 

### ISSUE THREAD

**Issue by pressureless** [tobguent/optimal-frames#1]
Hi,
Thanks for sharing the code of those excellent papers. I wonder is  the matrix m in objectivity reference frame invariance correct?
The paper  "Generic Objective Vortices for Flow Visualization" shows that for 2D, matrix M is as follows:
<img width="759" alt="image" src="https://user-images.githubusercontent.com/8511053/149437758-2e61ea74-46a2-4bc6-9676-11c74ac26a7e.png">

While in the source code, the position for the identity matrix is different:
<img width="1392" alt="image" src="https://user-images.githubusercontent.com/8511053/149437878-8d3dae29-98d9-45ca-a66d-1322ff49a285.png">

I didn't understand the paper well, please correct me if I missed something, thanks for the help.

**Comment by tobguent** [comment]
Hello!

Thanks for the message. In a linear problem, we can swap the columns of the system matrix, if we also swap the rows of the unknown vector in the same way. Here are system matrix M and unknown vector u from the paper:
![grafik](https://user-images.githubusercontent.com/4982169/149520003-db60af91-e03b-43c6-bfe5-44ce2e70d675.png)

In the code, we swapped the last columns of the system matrix, so the unknown vector also has its elements swapped:
![grafik](https://user-images.githubusercontent.com/4982169/149520219-d7511d0b-aaaf-49e8-9f54-0c4a8acface4.png)

The solution to the linear problem is the same. Since u3 and u4 are not needed for the computation of the new velocity field, this swap of the columns has no effect on the rest of the code. I just changed the expressions in the code to be equivalent to how they are phrased in the paper to avoid the confusion in the future.

Thanks for the very good question, and if you have any further feel free to send an email or post here.
